### ISSUE TITLE
Join config.json and protocol.json

### ISSUE THREAD

**Issue by shargon** [neo-project/neo-node#511]
What do you think about join this two configuration files in only one?

**Comment by lock9** [comment]
I think only one is better. Other fields may be optional.

**Comment by wanglongfei88** [comment]
Agree. I do not see too much advantage of having two separate configuration files. One config will be a better option.

**Comment by vncoelho** [comment]
I am not fully sure, in fact, they are a little bit different in terms of concepts.

But one config with the field `ProtocolConfiguration` also looks reasonable. I am not sure.
What @erikzhang thinks?

**Comment by erikzhang** [comment]
But we have plugin configuration files too. How to join all of them?

**Comment by shargon** [comment]
Plugins are optional, I prefer to keep them as now

**Comment by erikzhang** [comment]
Then I think there is no reason to join the config.json and protocol.json.

**Comment by vncoelho** [comment]
I think that I prefer not to join it right now as well, until we do not have a global state for configuration, which would be nice.

On the other hand, the idea @erikzhang gave of joining Plugin sounds great.
If Plugins are loaded it also looks from the same file. Then, we could later create tools for managing this file.
